tests/testsuite/rustflags.rs

```diff
@@ -1478,36 +1481,44 @@ fn env_rustflags_misspelled_build_script() {
 
 #[cargo_test]
 fn remap_path_prefix_ignored() {
-    // Ensure that --remap-path-prefix does not affect metadata hash.
-    let p = project().file("src/lib.rs", "").build();
-    p.cargo("build").run();
-    let rlibs = p
-        .glob("target/debug/deps/*.rlib")
-        .collect::<Result<Vec<_>, _>>()
-        .unwrap();
-    assert_eq!(rlibs.len(), 1);
-    p.cargo("clean").run();
-
-    let check_metadata_same = || {
-        let rlibs2 = p
-            .glob("target/debug/deps/*.rlib")
-            .collect::<Result<Vec<_>, _>>()
-            .unwrap();
-        assert_eq!(rlibs, rlibs2);
+    let get_c_metadata_re =
```

Comment:
Generally I think a regex worth a commment to explain itself.

Reply:
For very complex regexes, I can see that.  Maybe its a curse of knowledge thing but this seems relatively straightforward and I feel like any comment I write would just be repeating what the code does (extract two parameters from the CLI).

Reply:
Fair enough!

Will it be better that we adds extra leading space so flags are less likely to be captured from weird place like `my--crate-name`?
Also `+` to ensure we actually capture something?

(Surprisingly `rustc -C metadata= src/main.rs` works`)

```rust
        regex::Regex::new(r".* (--crate-name [^ ]+).* (-C ?metadata=[^ ]+).*").unwrap();
```